src/commands.ts

```diff
@@ -614,11 +621,147 @@ export const insertSection =
     return true
   }
 
+export const insertBackMatterSection =
+  (category: string) => (state: ManuscriptEditorState, dispatch?: Dispatch) => {
+    const bibliographySection = findChildrenByType(
+      state.doc,
+      schema.nodes.bibliography_section
+    )[0]
+    const backmatter = findChildrenByType(state.doc, schema.nodes.backmatter)[0]
+    let pos
+    // check if reference node exist to insert before it.
+    if (bibliographySection) {
+      pos = bibliographySection.pos
+    } else {
+      pos = backmatter.pos + backmatter.node.content.size + 1
+    }
+    const tr = state.tr.insert(
+      pos,
+      state.schema.nodes.section.createAndFill({
+        category: `MPSectionCategory:${category}`,
+      }) as SectionNode
+    )
+
+    if (dispatch) {
+      // place cursor inside section title
+      const selection = TextSelection.create(tr.doc, pos)
+      dispatch(tr.setSelection(selection).scrollIntoView())
+    }
+
+    return true
+  }
+
 const findSelectedList = findParentNodeOfType([
   schema.nodes.ordered_list,
   schema.nodes.bullet_list,
 ])
 
+export const insertAbstract = (
+  state: ManuscriptEditorState,
+  dispatch?: Dispatch
+) => {
+  if (
+    getMatchingChild(
+      state.doc,
+      (node) => node.attrs.category === 'MPSectionCategory:abstract',
+      true
+    )
+  ) {
+    return false
+  }
+  const abstracts = findChildrenByType(state.doc, schema.nodes.abstracts)[0]
+  // Insert abstract at the top of abstracts section
+  const pos = abstracts.pos + 1
+  const section = schema.nodes.section.createAndFill(
+    { category: 'MPSectionCategory:abstract' },
+    [
+      schema.nodes.section_title.create({}, schema.text('Abstract')),
+      schema.nodes.paragraph.create({ placeholder: 'Type abstract here...' }),
+    ]
+  ) as ManuscriptNode
+
+  const tr = state.tr.insert(pos, section)
+
+  if (dispatch) {
+    // place cursor inside section title
+    const selection = TextSelection.create(tr.doc, pos + 1)
+    dispatch(tr.setSelection(selection).scrollIntoView())
+  }
+  return true
+}
+export const insertContributors = (
+  state: ManuscriptEditorState,
+  dispatch?: Dispatch
+) => {
+  // Check if another contributors node already exists
+  if (getChildOfType(state.doc, schema.nodes.contributors, true)) {
+    return false
+  }
+
+  // Find the title node
+  const title = findChildrenByType(state.doc, state.schema.nodes.title)[0]
+
+  const pos = title.pos + title.node.nodeSize
+
+  const contributors = state.schema.nodes.contributors.create({
+    id: '',
+    firstInitials: true,
+  })
+  const affiliations = state.schema.nodes.affiliations.create({ id: '' })
+
+  const fragment = Fragment.fromArray([contributors, affiliations])
+
+  const tr = state.tr.insert(pos, fragment)
+
+  if (dispatch) {
+    const selection = TextSelection.create(tr.doc, pos + 1)
```

Comment:
I tried the nodeSelection but with no luck since the selection will be on the contributors node wrapper

Reply:
Not sure what you mean. Each node view can have a `selectNode` function that allows us to customize the behavior when the node is selected. Does that not work?

Reply:
Thinking about it more, maybe we should open the popup directly without creating a node, and if the user decides to add an author, the node will be created and the view will render.

Reply:
> Not sure what you mean. Each node view can have a `selectNode` function that allows us to customize the behavior when the node is selected. Does that not work?

The issue here that the edit authors button is injected after the creation of the node, so it doesn't exist as a node inside the editorView doc object. and selecting the contributors node won't trigger the popup since the event listener is attached only on the button.

Reply:
> Thinking about it more, maybe we should open the popup directly without creating a node, and if the user decides to add an author, the node will be created and the view will render.

Yes, this sound good.

but we need to move the logic into the React app in order to benefit form the state and re-render to trigger the node insertion when there is any authors inserted. what do you think?

Reply:
Additionally, the current popup is injected inside the contributors node and relies on both the contributors and affiliations node data. Using an attribute would be a more optimal solution given the current implementation of the contributors node functionality. Otherwise, we would need to introduce multiple changes to achieve your suggestion.

<img width="967" alt="Screenshot 2024-06-23 at 11 46 08 AM" src="https://github.com/Atypon-OpenSource/manuscripts-body-editor/assets/82392169/915f1219-87e7-433d-92dc-1a2c1d1ce59d">


Could you please let me know if there are any downsides to adding a new attribute for the contributors node?

Reply:
Done, implemented using the selectNode